video/out/cocoa_common.m

```diff
@@ -873,4 +883,16 @@ - (void)windowDidBecomeKey:(NSNotification *)notification
     [self didChangeMousePosition];
 }
 
+- (void)windowDidMiniaturize:(NSNotification *)notification
+{
+    struct vo_cocoa_state *s = self.vout->cocoa;
+    s->pending_events |= VO_EVENT_WIN_STATE;
+}
+
+- (void)windowDidDeminiaturize:(NSNotification *)notification
```

Comment:
He is calling it from `window.m`, so it's correct. It's rather subtle but this the correct way to do it. The adapter implements the `NSWindowDelegate` protocol, and is the delegate class of the fullscreen borderless window when going fullscreen.